File: app/src/main/java/com/ghostnexora/vpn/tunnel/strategy/TunnelStrategy.java
package com.ghostnexora.vpn.tunnel.strategy;

import com.ghostnexora.vpn.data.model.ConnectionMode;
import com.ghostnexora.vpn.data.model.VpnProfile;
import com.jcraft.jsch.Session;

/** Java contract for SSH-backed tunnel strategies. */
public interface TunnelStrategy {
    boolean supports(ConnectionMode mode);
    Session connect(VpnProfile profile);
    void disconnect(Session session);
}

File: app/src/main/java/com/ghostnexora/vpn/tunnel/strategy/TunnelStrategy.kt
package com.ghostnexora.vpn.tunnel.strategy

import com.ghostnexora.vpn.data.model.ConnectionMode
import com.ghostnexora.vpn.data.model.VpnProfile
import com.jcraft.jsch.Session

interface TunnelStrategy {
    fun supports(mode: ConnectionMode): Boolean
    fun connect(profile: VpnProfile): Session
    fun disconnect(session: Session?)
}
